jailer/src/lib.rs

```diff
@@ -97,6 +96,122 @@ pub enum Error {
     Write(PathBuf, io::Error),
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        match *self {
+            Canonicalize(ref path, ref io_err) => {
+                write!(f, "Failed to canonicalize path {:?}: {:?}", path, io_err)
+            }
+            CgroupInheritFromParent(ref path, ref filename) => write!(
+                f,
+                "Failed to inherit cgroups configurations from file {} in path {:?}",
+                filename, path
+            ),
+            CgroupLineNotFound(ref proc_mounts, ref controller) => write!(
+                f,
+                "{} configurations not found in {}",
+                controller, proc_mounts
+            ),
+            CgroupLineNotUnique(ref proc_mounts, ref controller) => write!(
+                f,
+                "Found more than one cgroups configuration line in {} for {}",
+                proc_mounts, controller
+            ),
+            ChangeDevNetTunOwner(ref err) => {
+                write!(f, "Failed to change owner for /dev/net/tun: {}", err)
+            }
+            Chroot(ref err) => write!(f, "Failed to chroot into the new jail folder: {}", err),
+            CloseNetNsFd(ref err) => write!(f, "Failed to close netns fd: {}", err),
+            CloseDevNullFd(ref err) => write!(f, "Failed to close /dev/null fd: {}", err),
+            Copy(ref file, ref path, ref err) => write!(
+                f,
+                "Failed to copy {:?} to {:?}: error {:?}",
```

Comment:
```suggestion
                "Failed to copy {:?} to {:?}: {}",
```
to be consistent with other error messages.

Reply:
The inner error is an `io::Error`: `Copy(PathBuf, PathBuf, io::Error)` - which doesn't implement `Display`, so formatting with `{}` wont't work.